packages/apollo-federation/src/composition/compose.ts

```diff
@@ -419,6 +420,38 @@ export function composeServices(services: ServiceDefinition[]) {
     keyDirectivesMap,
   } = buildMapsFromServiceList(services);
 
+  for (const [typeName, definitions] of Object.entries(definitionsMap)) {
```

Comment:
This is a cleanup pass over the `definitionsMap`. It's possible this could be done as the map is built up, which would probably be preferable but might require reshaping the map's type in order to do so.

Reshaping the map and adding some specific information on the federation metadata w.r.t. value types (i.e. an `isValueType` field) would be a more expressive way of resolving the owning service during `buildQueryPlan` as well, as opposed to nulling out `serviceName` and then falling back to `parentGroup.serviceName` in `buildQueryPlan`.